relay-general/src/user_agent.rs

```diff
@@ -126,6 +126,31 @@ impl RawUserAgentInfo<String> {
 }
 
 impl<'a> RawUserAgentInfo<&'a str> {
+    pub fn populate_event_headers(&self, headers: &mut Headers) {
+        let mut insert_header = |key: &str, val: Option<&str>| {
+            if let Some(val) = val {
+                if !headers.contains(key) {
+                    headers.insert(HeaderName::new(key), Annotated::new(HeaderValue::new(val)));
+                }
+            }
+        };
+
+        insert_header(RawUserAgentInfo::USER_AGENT, self.user_agent);
+        insert_header(
+            ClientHints::SEC_CH_UA_PLATFORM,
+            self.client_hints.sec_ch_ua_platform,
+        );
+        insert_header(
+            ClientHints::SEC_CH_UA_PLATFORM_VERSION,
+            self.client_hints.sec_ch_ua_platform_version,
+        );
+        insert_header(ClientHints::SEC_CH_UA, self.client_hints.sec_ch_ua);
+        insert_header(
+            ClientHints::SEC_CH_UA_MODEL,
+            self.client_hints.sec_ch_ua_model,
+        );
+    }
```

Comment:
Can this function only operator on `&'a str`? If not, I would put it in the generic `impl<S: ...>` block.

Reply:
it relies on &str being copy, do you think I should make it generic and clone it instead ?

Reply:
What if you call `as_ref` or `as_deref` on the values you pass into `insert_header`?

Reply:
ok neat, it works that way yeah!